src/screens/Game/GameSubMenu/index.tsx

```diff
@@ -271,16 +263,19 @@ export default function GamesSubmenu({
                 {t('submenu.addShortcut', 'Add shortcut')}
               </button>
             )}
-            {isLinux && !eosOverlayInstalling && (
-              <button
-                className="link button is-text is-link"
-                onClick={handleEosOverlay}
-              >
-                {eosOverlayEnabled
-                  ? t('submenu.disableEosOverlay', 'Disable EOS Overlay')
-                  : t('submenu.enableEosOverlay', 'Enable EOS Overlay')}
-              </button>
-            )}
+            {isLinux &&
+              (eosOverlayRefresh ? (
+                refreshCircle()
+              ) : (
+                <button
+                  className="link button is-text is-link"
+                  onClick={async () => handleEosOverlay()}
+                >
+                  {eosOverlayEnabled
+                    ? t('submenu.disableEosOverlay', 'Disable EOS Overlay')
+                    : t('submenu.enableEosOverlay', 'Enable EOS Overlay')}
+                </button>
+              ))}
```

Comment:
I think it would make it easier to read/maintain if we extract all this markup and logic into an `EosOverlayToggle` or something like that.

Maybe not for this PR but I think this component is pretty complex and we could extract each element in the list into it's own component for readability.

Reply:
We need a general component that download/disable/enable a tool and show progress.
Actually we need something like wine-manager but for runtimes, anti-cheat, etc.